src/Operators/finitedifference.jl

```diff
@@ -1459,6 +1459,139 @@ end
     return Geometry.Contravariant3Vector(vᶠ * a)
 end
 
+"""
+    U = FCTBorisBook(;boundaries)
+    U.(v, x)
+
+Correct the flux using the flux-corrected transport formulation by Boris and Book
+Input arguments:
+- a face-valued vector field `v`
+- a center-valued field `x`
+```math
+U(v,x)[i] = \\begin{cases}
+  v[i] \\left(-2 x[i-\\tfrac{3}{2}] + 10 x[i-\\tfrac{1}{2}] + 4 x[i+\\tfrac{1}{2}] \\right) / 12  \\textrm{, if } v[i] > 0 \\\\
+  v[i] \\left(4 x[i-\\tfrac{1}{2}] + 10 x[i+\\tfrac{1}{2}] -2 x[i+\\tfrac{3}{2}]  \\right) / 12  \\textrm{, if } v[i] < 0
+  \\end{cases}
+```
+This stencil is based on [BorisBook1973](@cite).
+
+Supported boundary conditions are:
+- [`ThirdOrderOneSided(x₀)`](@ref): uses the third-order downwind reconstruction to compute `x` on the left boundary,
+and the third-order upwind reconstruction to compute `x` on the right boundary.
+
+!!! note
+    Similar to the [`Upwind3rdOrderBiasedProductC2F`](@ref) operator, these boundary conditions do not define the value at the actual boundary faces, and so this operator cannot be materialized directly: it needs to be composed with another operator that does not make use of this value, e.g. a [`DivergenceF2C`](@ref) operator, with a [`SetValue`](@ref) boundary.
+"""
+struct FCTBorisBook{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+FCTBorisBook(; kwargs...) = FCTBorisBook(NamedTuple(kwargs))
+
+return_eltype(::FCTBorisBook, V, A) =
+    Geometry.Contravariant3Vector{eltype(eltype(V))}
+
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceFiniteDifferenceSpace,
+    arg_space::Spaces.CenterFiniteDifferenceSpace,
+) = velocity_space
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceExtrudedFiniteDifferenceSpace,
+    arg_space::Spaces.CenterExtrudedFiniteDifferenceSpace,
+) = velocity_space
+
+@inline function fct_boris_book(v, a⁻, a⁺, a⁺⁺, a⁺⁺⁺, step)
+    if v != zero(eltype(a⁻))
+        sign(v) ⊠ (RecursiveApply.rmap(
+            max,
+            zero(eltype(a⁻)),
+            RecursiveApply.rmap(
```

Comment:
I think we can improve this a bit-- I need a bit of time to think about this, but the compiler may have trouble with the very nested `rmap`s.

A more important concern is about the `v != zero(eltype(a⁻))` comparison. Let's consider how this comparison behaves:

```julia
julia> compare(num::FT) where {FT} = num ≠ FT(0)
compare (generic function with 1 method)

julia> compare(Float32(1e-7))
true

julia> compare(Float32(1e-15))
true

julia> compare(Float32(1e-60))
false

julia> compare(Float32(1e-30))
true

julia> compare(Float64(1e-30))
true

julia> compare(Float64(1e-300))
true

julia> compare(Float64(1e-600))
false
```

Is this what we want?

Reply:
Interesting. I saw that this was used in several other places, e.g., the construction of a [zero field](https://github.com/CliMA/ClimaCore.jl/blob/b133c3ae65e40c993f46c57466563d8cd1f99faa/src/Fields/Fields.jl#L212) and I thought that this was indeed the type-safe way to do this

Reply:
Discussed this, looks good because the only difference is based on `sign`